pyfr/plugins/integrate.py

```diff
@@ -143,23 +151,11 @@ def _eval_exprs(self, intg):
             subs = dict(zip(pnames, psolns))
             subs.update(zip('xyz', plocs))
 
-            # Compute any required gradients
+            # Prepare any required gradient
             if self._gradpnames:
```

Comment:
I think we can remove this check.

Reply:
I think that this is needed in order to compute all gradients before looping over the regions and elements. This could maybe be avoided if we passed the region information to `_grad_pvars`